src/log/log_sys.c

```diff
@@ -31,8 +31,7 @@ __wt_log_system_backup_id(WT_SESSION_IMPL *session)
      * If we're not logging or incremental backup isn't turned on or this version doesn't support
      * the system log record, we're done.
      */
-    if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED) ||
-      !FLD_ISSET(conn->log_flags, WT_CONN_LOG_INCR_BACKUP))
+    if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED))
```

Comment:
This is a spot I wasn't sure about. I don't completely understand the original logic - the ticket that added it doesn't appear to be specific to log-based incremental backup, but it seems we need both of these flags to be set for us to generate the new log record.

Is it possible I misunderstood the `WT_CONN_LOG_INCR_BACKUP` flag and that this flag actually means "log that we did an incremental backup"?

Reply:
Yes, per my earlier comment, this flag is whether to log the block-based IDs or not. This all needs to stay

Reply:
Thanks for confirming - reinstated in d624f4f.